src/parse/parse-token-request.ts

```diff
@@ -292,7 +291,7 @@ const parseClientAssertion = (
   }
 
   return {
-    header,
+    header: { ...header, alg: header.alg },
```

Comment:
what is this doing?

Reply:
Was having some weird type issues where 
```
Type 'ProtectedHeaderParameters' is not assignable to type 'ClientAssertionHeaders'.
  Types of property 'alg' are incompatible.
    Type 'string | undefined' is not assignable to type 'string'.
      Type 'undefined' is not assignable to type 'string'
```

It's odd because the type should narrow given we've got the check `if(!header.alg)` so it should satisfy the type. Also weirdly, having this return ` header: {  alg: header.alg }` is fine. The idea here was to satisfy the return type but also preserving any other fields in header by spreading the it. Given I've typed this many words, I might just cast they type instead 😅

Reply:
makes sense!